app/lib/user_annotation_service.rb

```diff
@@ -0,0 +1,54 @@
+class UserAnnotationService
+  # Methods to interact with annotation data, beyond just visualization
+
```

Comment:
this function could use some documentation on what these params are, especially the structure expected by user_data_arrays_attributes and loaded_annotation .  I *think* I know what they mean, but I'm not sure.

Reply:
Done in 69caefc1.